scoring.js

```diff
@@ -0,0 +1,43 @@
+const calculateScore = (player) => {
+
+  switch (player.position) {
+    case 'QB':
+      return passingScore() + rushingScore()
+    case 'RB':
+    case 'WR':
+      return rushingScore() + kickReturnScore() + puntReturnScore() + receivingScore()
+    case 'TE':
+      return receivingScore()
+    default:
+      return 0
+  }
+
+
+  function passingScore() {
+    const passingScore = (player.stats.passing.yards / 25) + (player.stats.passing.touchdowns * 6) + (player.stats.passing.interceptions * -3)
+    return passingScore
+  }
+
+  function rushingScore() {
+    const rushingScore = (player.stats.rushing.yards / 10) + (player.stats.rushing.touchdowns * 6) + (player.stats.rushing.fumbles * -3)
+    return rushingScore
+  }
+
+  function receivingScore() {
+    const receivingScore = player.stats.receiving.receptions + (player.stats.receiving.yards / 10) + (player.stats.receiving.touchdowns * 6) + (player.stats.receiving.fumbles * -3)
+    return receivingScore
+  }
```

Comment:
Since you are one lining this calculation there is no reason to put it in a variable just to return the variable.

```JavaScript
function receivingScore(player) {
  return player.stats.receiving.receptions + (player.stats.receiving.yards / 10) + (player.stats.receiving.touchdowns * 6) + (player.stats.receiving.fumbles * -3)
}
```

Though since this is a decently big calculation, it might make sense to put each into a variable and then return the final sum up, makes it more readable.

```JavaScript
function calculateReceivingScore(player) {
  const receptions = player.stats.receiving.receptions
  const yards = player.stats.receiving.yards / 10
  const touchdowns = player.stats.receiving.touchdowns * 6
  const fumbles = player.stats.receiving.fumbles * -3

  return receptions + yards + touchdowns + fumbles
}
```